### PR TITLE
Update CI to work with Xcode 16 / iOS 18

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes APP-####

## What changed (plus any additional context for devs)
- Bump detox to latest version
- Now that all CI infra is ready for Xcode 16 / iOS 18 we can bump the selected sim for detox to use `iPhone 16 pro`
- Also removed eslint caching to avoid issues in CI
- Fix failing e2e

## Screen recordings / screenshots
I can't add anything but CI should be green

## What to test
Nothing really other than make sure all the CI steps are passing

### CODE REVIEW

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/bump-detox-sim/7381e7105b4435846ad2696b8da60daa2d84d490.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/bump-detox-sim/7381e7105b4435846ad2696b8da60daa2d84d490.ipa) for 7381e7105b4435846ad2696b8da60daa2d84d490

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/bump-detox-sim/3cc90947b70c0e9f2159d241b677abfe02273351.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/bump-detox-sim/3cc90947b70c0e9f2159d241b677abfe02273351.ipa) for 3cc90947b70c0e9f2159d241b677abfe02273351

File: e2e/9_swaps.spec.ts [hunk 1]
```diff
@@ -62,9 +62,7 @@ describe('Swap Sheet Interaction Flow', () => {
     await tap('swap-button');
     await delayTime('very-long');
 
-    // flaky
-    // await swipeUntilVisible('token-to-buy-dai-1', 'token-to-buy-list', 'up', 100);
-    await swipe('token-to-buy-list', 'up', 'slow', 0.2);
+    await swipeUntilVisible('token-to-buy-dai-1', 'token-to-buy-list', 'up', 100);
```
Comment (walmat) [inline e2e/9_swaps.spec.ts:65]: did you verify this works as expected now?

File: e2e/helpers.ts [hunk 2]
```diff
@@ -201,6 +202,7 @@ export async function clearField(elementId: string | RegExp) {
 
 export async function tapAndLongPress(elementId: string | RegExp, duration?: number) {
   try {
+    // @ts-ignore
```
Comment (walmat) [inline e2e/helpers.ts:205]: can you use `@ts-expect-error` instead?

File: e2e/helpers.ts [hunk 3]
```diff
@@ -209,6 +211,7 @@ export async function tapAndLongPress(elementId: string | RegExp, duration?: num
 
 export async function tapAndLongPressByText(text: string | RegExp, duration?: number) {
   try {
+    // @ts-ignore
```
Comment (walmat) [inline e2e/helpers.ts:214]: can you use `@ts-expect-error` instead?

**Review by walmat — COMMENTED** [review]
nits, but question about the swipe util and if it's consistently working now. Before it was not.

File: e2e/9_swaps.spec.ts (hunk 1 above)
Comment (brunobar79) [reply e2e/9_swaps.spec.ts:65]: all the runs before thsi tests were failing. This is passing, so I'd say yeah

File: e2e/helpers.ts (hunk 2 above)
Comment (brunobar79) [reply e2e/helpers.ts:205]:
```suggestion
    // @ts-expect-error
```

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/bump-detox-sim/be77d65c9ccdf072c1768194e59e30b5c475c06c.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/bump-detox-sim/be77d65c9ccdf072c1768194e59e30b5c475c06c.ipa) for be77d65c9ccdf072c1768194e59e30b5c475c06c

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/bump-detox-sim/5d723ef5a14f57b1734039d0a8611ea5511ee3c0.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/bump-detox-sim/5d723ef5a14f57b1734039d0a8611ea5511ee3c0.ipa) for 5d723ef5a14f57b1734039d0a8611ea5511ee3c0

**Review by walmat — APPROVED** [review]

**Review by christianbaroni — APPROVED** [review]

**Pull request merged**